pkg/cidr/cidr.go

```diff
@@ -33,6 +33,25 @@ type CIDR struct {
 	*net.IPNet
 }
 
+func (in *CIDR) DeepEqual(other *CIDR) bool {
+	if other == nil {
```

Comment:
Shouldn't we also check for `in` being nil here (Like in `CiliumNetworkPolicy` below)?

Reply:
It's up to the caller to check for the receiver to be nil.